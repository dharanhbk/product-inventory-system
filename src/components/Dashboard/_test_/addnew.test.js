import React from 'react';
import ReactDOM from 'react-dom'
import {MemoryRouter as Router} from 'react-router-dom'
import { render  } from '@testing-library/react';
import { mount, configure , shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import '@testing-library/jest-dom/extend-expect'
import AddNew from '../addNew';


configure({adapter:new Adapter()});

test('should test Addnewcomponent', () => {
  const wrapper = shallow(<AddNew />);
  expect(wrapper).toMatchSnapshot();
 });


describe("all test for Add new component", ()=>{
    it('renders the Add new component without crashing!', ()=>{
      const div = document.createElement('div')
      ReactDOM.render(<Router><AddNew></AddNew></Router>, div)
    })
  })

  it('check if button renders in correct way', ()=>{
    const {getByTestId} = render(<Router><AddNew></AddNew></Router>)
    expect(getByTestId('addnewbutton')).toHaveTextContent('Save Item')
})
it(" check product name", () => {
  const wrapper = mount(<input type="text" placeholder="Product name" />);
  const input = wrapper.find("input");
  expect(input.prop("type")).toEqual("text");
  expect(input.prop("placeholder")).toEqual("Product name" );
  expect(input).toHaveLength(1)
});
it(" check product quantity", () => {
  const wrapper = mount(<input type="text" placeholder="Quantity" />);
  const input = wrapper.find("input");
  expect(input.prop("type")).toEqual("text");
  expect(input.prop("placeholder")).toEqual("Quantity" );
  expect(input).toHaveLength(1)
});
