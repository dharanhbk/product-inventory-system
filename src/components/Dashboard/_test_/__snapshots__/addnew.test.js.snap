// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Addnewcomponent 1`] = `ShallowWrapper {}`;
